Sources/GRPC/AsyncAwaitSupport/GRPCAsyncRequestStream.swift

```diff
@@ -21,40 +21,64 @@
 @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
 public struct GRPCAsyncRequestStream<Element: Sendable>: AsyncSequence {
   @usableFromInline
-  internal typealias _WrappedStream = PassthroughMessageSequence<Element, Error>
+  enum Backing: Sendable {
+    case passthroughMessageSequence(PassthroughMessageSequence<Element, Error>)
+    case asyncStream(AsyncThrowingStream<Element, Error>)
+  }
 
   @usableFromInline
-  internal let _stream: _WrappedStream
+  internal let backing: Backing
 
   @inlinable
-  internal init(_ stream: _WrappedStream) {
-    self._stream = stream
+  internal init(_ sequence: PassthroughMessageSequence<Element, Error>) {
+    self.backing = .passthroughMessageSequence(sequence)
+  }
+
+  @inlinable
+  public init(_ stream: AsyncThrowingStream<Element, Error>) {
```

Comment:
Escaping the continuation from the stream is probably the most common way this will be used which is quite an obtuse API. 

I think we'd be better served by hiding `AsyncThrowingStream` from the API and having a `static` `func` returning a _wrapped_ `AsyncThrowingStream.Continuation`. Alternatively it could just be a wrapped `NIOAsyncSequenceProducer.Source` with a no-op delegate and no-op back pressure strategy (we could always swap them out later).